backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -234,6 +234,111 @@ public CommitEntity getCommitEntity(
     return session.load(CommitEntity.class, commitHash);
   }
 
+  @Override
+  public boolean deleteCommits(
+      RepositoryIdentification repositoryIdentification,
+      List<String> commitShas,
+      RepositoryDAO repositoryDAO)
+      throws ModelDBException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      session.beginTransaction();
+      RepositoryEntity repositoryEntity =
+          repositoryDAO.getRepositoryById(session, repositoryIdentification, true);
+
+      Query<CommitEntity> getCommitQuery =
+          session.createQuery(
+              "From "
+                  + CommitEntity.class.getSimpleName()
+                  + " c WHERE c.commit_hash IN (:commitHashes)",
+              CommitEntity.class);
+      getCommitQuery.setParameter("commitHashes", commitShas);
+      List<CommitEntity> commitEntities = getCommitQuery.getResultList();
+
+      for (CommitEntity commitEntity : commitEntities) {
+        if (!commitEntity.getChild_commits().isEmpty()) {
+          throw new ModelDBException(
+              "Commit has the child, please delete child commit first", Code.FAILED_PRECONDITION);
+        }
+      }
+
+      String getBranchByCommitHQLBuilder =
+          "FROM "
+              + BranchEntity.class.getSimpleName()
+              + " br where br.id.repository_id = :repositoryId "
+              + " AND br.commit_hash IN (:commitHashes) ";
+      Query<BranchEntity> getBranchByCommitQuery =
+          session.createQuery(getBranchByCommitHQLBuilder, BranchEntity.class);
+      getBranchByCommitQuery.setParameter("repositoryId", repositoryEntity.getId());
+      getBranchByCommitQuery.setParameter("commitHashes", commitShas);
+      List<BranchEntity> branchEntities = getBranchByCommitQuery.list();
+      if (branchEntities != null && !branchEntities.isEmpty()) {
+        StringBuilder errorMessage = new StringBuilder("Commit is associated with branch name : ");
+        int count = 0;
+        for (BranchEntity branchEntity : branchEntities) {
+          errorMessage.append(branchEntity.getId().getBranch());
+          if (count < branchEntities.size() - 1) {
+            errorMessage.append(", ");
+          }
+          count++;
+        }
+        throw new ModelDBException(errorMessage.toString(), Code.FAILED_PRECONDITION);
+      }
+
+      String getLabelsHql =
```

Comment:
deleting commit with label should be fine, when deleting the commit we need to delete the associated label mapping.
if there is some prior discussion I am missing do let me know

Reply:
fixed